File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/viewmodel/activity/ResultVModel.java
package io.ganguo.demo.viewmodel.activity;

import android.app.Activity;
import android.content.Intent;
import android.view.View;

import io.ganguo.demo.R;
import io.ganguo.demo.bean.Constants;
import io.ganguo.demo.databinding.ActivityResultBinding;
import io.ganguo.viewmodel.common.HeaderItemViewModel;
import io.ganguo.viewmodel.common.HeaderViewModel;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.ViewModelHelper;
import io.ganguo.library.ui.view.ActivityInterface;
import io.ganguo.utils.util.Randoms;

/**
 * Activity ResultVModel
 * Created by Roger on 6/7/16.
 */
public class ResultVModel extends BaseViewModel<ActivityInterface<ActivityResultBinding>> {

    public String randomLetter = Randoms.getRandomCapitalLetters(20);

    @Override
    public void onViewAttached(View view) {
    }


    /**
     * function:create Random Click
     *
     * @return
     */
    public View.OnClickListener onCrateRandomClick() {
        return v -> {
            Intent intent = getView().getActivity().getIntent();
            intent.putExtra(Constants.DATA, randomLetter);
            getView().getActivity().setResult(Activity.RESULT_OK, intent);
            getView().getActivity().finish();
        };
    }

    @Override
    public int getItemLayoutId() {
        return R.layout.activity_result;
    }

}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/viewmodel/activity/ResultVModel.kt
package io.ganguo.demo.viewmodel.activity

import android.app.Activity
import android.view.View

import io.ganguo.demo.R
import io.ganguo.demo.bean.Constants
import io.ganguo.demo.databinding.ActivityResultBinding
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.core.viewinterface.ActivityInterface
import io.ganguo.utils.util.Randoms

/**
 * Activity ResultVModel
 * Created by Roger on 6/7/16.
 */
class ResultVModel : BaseViewModel<ActivityInterface<ActivityResultBinding>>() {
    override val layoutId: Int by lazy { R.layout.activity_result }
    var randomLetter: String = Randoms.getRandomCapitalLetters(20)


    override fun onViewAttached(view: View) {}


    /**
     * Random Click
     *
     * @return
     */
    fun actionRandomClick(): View.OnClickListener {
        return View.OnClickListener {
            val intent = viewInterface.activity.intent
            intent.putExtra(Constants.DATA, randomLetter)
            viewInterface.activity.setResult(Activity.RESULT_OK, intent)
            viewInterface.activity.finish()
        }
    }


}
